fv/utils/utils.go

```diff
@@ -212,92 +206,32 @@ func IPSetNameForSelector(ipVersion int, rawSelector string) string {
 	return ipVerConf.NameForMainIPSet(setID)
 }
 
-// Run a connection test command.
-// Report if connection test is successful and packet loss string for packet loss test.
-func RunConnectionCmd(connectionCmd *exec.Cmd, logMsg string) *connectivity.Result {
-	outPipe, err := connectionCmd.StdoutPipe()
-	Expect(err).NotTo(HaveOccurred())
-	errPipe, err := connectionCmd.StderrPipe()
-	Expect(err).NotTo(HaveOccurred())
-	err = connectionCmd.Start()
-	Expect(err).NotTo(HaveOccurred())
-
-	var wg sync.WaitGroup
-	wg.Add(2)
-	var wOut, wErr []byte
-	var outErr, errErr error
-
-	go func() {
-		defer wg.Done()
-		wOut, outErr = ioutil.ReadAll(outPipe)
-	}()
-
-	go func() {
-		defer wg.Done()
-		wErr, errErr = ioutil.ReadAll(errPipe)
-	}()
-
-	wg.Wait()
-	Expect(outErr).NotTo(HaveOccurred())
-	Expect(errErr).NotTo(HaveOccurred())
-
-	err = connectionCmd.Wait()
-
-	log.WithFields(log.Fields{
-		"stdout": string(wOut),
-		"stderr": string(wErr)}).WithError(err).Info(logMsg)
+// HasSyscallConn represents objects that can return a syscall.RawConn
+type HasSyscallConn interface {
+	SyscallConn() (syscall.RawConn, error)
+}
 
+// ConnMTU returns the MTU of the connection for _connected_ connections. That
+// excludes unconnected udp which requires different approach for each peer.
+func ConnMTU(hsc HasSyscallConn) (int, error) {
+	c, err := hsc.SyscallConn()
 	if err != nil {
-		return nil
+		return 0, err
 	}
 
-	r := regexp.MustCompile(`RESULT=(.*)\n`)
-	m := r.FindSubmatch(wOut)
-	if len(m) > 0 {
-		var resp connectivity.Result
-		err := json.Unmarshal(m[1], &resp)
-		if err != nil {
-			log.WithError(err).WithField("output", string(wOut)).Panic("Failed to parse connection check response")
-		}
-		return &resp
-	}
-	return nil
-}
-
-const ConnectionTypeStream = "stream"
-const ConnectionTypePing = "ping"
-
-type ConnConfig struct {
-	ConnType string
-	ConnID   string
-}
-
-func (cc ConnConfig) getTestMessagePrefix() string {
-	return cc.ConnType + ":" + cc.ConnID + "~"
-}
-
-// Assembly a test message.
-func (cc ConnConfig) GetTestMessage(sequence int) connectivity.Request {
-	req := connectivity.NewRequest(cc.getTestMessagePrefix() + fmt.Sprintf("%d", sequence))
-	return req
-}
+	mtu := 0
+	var sysErr error
+	err = c.Control(func(fd uintptr) {
+		mtu, sysErr = syscall.GetsockoptInt(int(fd), syscall.IPPROTO_IP, syscall.IP_MTU)
+	})
 
-// Extract sequence number from test message.
-func (cc ConnConfig) GetTestMessageSequence(msg string) (int, error) {
-	msg = strings.TrimSpace(msg)
-	seqString := strings.TrimPrefix(msg, cc.getTestMessagePrefix())
```

Comment:
I moved this to connectivity to avoid an import cycle and it also feels like it belongs there now